### ISSUE TITLE
banana 1.5 - heatmap Solr version dependency?

### ISSUE THREAD

**Issue by dougmcclure** [lucidworks/banana#108]
I've tried the best I could to get the heatmap panel to work against my Solr 4.7.1.  Does this panel need a newer version (eg 4.10 with some Pivot Facet improvements)?

Tks!

**Comment by tarek-salah** [comment]
@dougmcclure I tested it with Solr 4.7.0 and it works fine for me 
did you check your configuration for the panel?

**Comment by dougmcclure** [comment]
Yes, I've tried to use this a dozen different times with no luck rendering
anything. Are there any requirements for what field types are selected for
"Row Field" and "Column Field"?  I'm most used to having a count type
metric to drive the value for heatmap coloring.

**Comment by tarek-salah** [comment]
@dougmcclure  What are the types of fields that you are using for "Row Field" and "Column Field"?

**Comment by dougmcclure** [comment]
For the two fields I've added in my last attempt, the schema shows this:

dynamicField name="*_ds" type="tdouble" indexed="true" stored="true" required="false" multiValued="false" docValues="true"

fieldType name="tdouble" class="solr.TrieDoubleField" precisionStep="8" positionIncrementGap="0"

Doug

**Comment by dougmcclure** [comment]
Any insights into this yet?

**Comment by nickvasilyev** [comment]
Is your collection sharded?

**Comment by dougmcclure** [comment]
Yes, each collection has two shards.

**Comment by nickvasilyev** [comment]
Try adding distrib=false to heat map query and see if that helps.

**Comment by dougmcclure** [comment]
That got me to a grid, but nothing appears to be showing up in a given cell. I'm expecting seeing some number value and shading representing count.

**Comment by nickvasilyev** [comment]
Alright, that's progress. See what query it is running and run it manually
in sole to make sure you are getting data back.

**Comment by dougmcclure** [comment]
Ok, ran the query in browser and here's a sample of the results returned. Looks like I have some results which should be plotting.

{
  "responseHeader":{
    "status":0,
    "QTime":6,
    "params":{
      "facet":"true",
      "indent":"true",
      "q":"_:_",
      "facet.limit":"10",
      "distrib":"false",
      "facet.pivot.mincount":"0",
      "wt":["json",
        "json"],
      "facet.pivot":"collectionName_ts,numberSuccessful_ds",
      "fq":["timestamp:[NOW/MINUTE-5MINUTE TO NOW/MINUTE+1MINUTE]",
        "_datasource:\"S-STATS-153\""],
      "rows":"0"}},
  "response":{"numFound":26,"start":0,"docs":[]
  },
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_dates":{},
    "facet_ranges":{},
    "facet_pivot":{
      "collectionName_ts,numberSuccessful_ds":[{
          "field":"collectionName_ts",
          "value":"GRStats-Collecti",
          "count":13,
          "pivot":[{
              "field":"numberSuccessful_ds",
              "value":2.0,
              "count":12},
            {
              "field":"numberSuccessful_ds",
              "value":3.0,
              "count":1},
            {
              "field":"numberSuccessful_ds",
              "value":0.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":1.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":4.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":5.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":6.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":7.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":8.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":9.0,
              "count":0}]},
        {
          "field":"collectionName_ts",
          "value":"GRLogs-Collectio",
          "count":12,
          "pivot":[{
              "field":"numberSuccessful_ds",
              "value":8.0,
              "count":12},
            {
              "field":"numberSuccessful_ds",
              "value":0.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":1.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":2.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":3.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":4.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":5.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":6.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":7.0,
              "count":0},
            {
              "field":"numberSuccessful_ds",
              "value":9.0,
              "count":0}]},

**Comment by nickvasilyev** [comment]
Ok, hover over the blocks that should have values and see if any data shows
up in tooltip.

**Comment by dougmcclure** [comment]
I see the tooltip, all values are (0). None of the counts show up like they do for the ones with results in the test query.

**Comment by karimfikry** [comment]
I tried your attached data which is:

``` json
{
  "responseHeader": {
    "status": 0,
    "QTime": 6,
    "params": {
      "facet": "true",
      "indent": "true",
      "q": ":",
      "facet.limit": "10",
      "distrib": "false",
      "facet.pivot.mincount": "0",
      "wt": [
        "json",
        "json"
      ],
      "facet.pivot": "collectionName_ts,numberSuccessful_ds",
      "fq": [
        "timestamp:[NOW/MINUTE-5MINUTE TO NOW/MINUTE+1MINUTE]",
        "_datasource:\"S-STATS-153\""
      ],
      "rows": "0"
    }
  },
  "response": {
    "numFound": 26,
    "start": 0,
    "docs": []
  },
  "facet_counts": {
    "facet_queries": {},
    "facet_fields": {},
    "facet_dates": {},
    "facet_ranges": {},
    "facet_pivot": {
      "collectionName_ts,numberSuccessful_ds": [
        {
          "field": "collectionName_ts",
          "value": "GRStats-Collecti",
          "count": 13,
          "pivot": [
            {
              "field": "numberSuccessful_ds",
              "value": 2,
              "count": 12
            },
            {
              "field": "numberSuccessful_ds",
              "value": 3,
              "count": 1
            },
            {
              "field": "numberSuccessful_ds",
              "value": 0,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 1,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 4,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 5,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 6,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 7,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 8,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 9,
              "count": 0
            }
          ]
        },
        {
          "field": "collectionName_ts",
          "value": "GRLogs-Collectio",
          "count": 12,
          "pivot": [
            {
              "field": "numberSuccessful_ds",
              "value": 8,
              "count": 12
            },
            {
              "field": "numberSuccessful_ds",
              "value": 0,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 1,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 2,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 3,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 4,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 5,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 6,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 7,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 9,
              "count": 0
            }
          ]
        },
                {
          "field": "collectionName_ts",
          "value": "GRLogs-Collectio",
          "count": 1,
          "pivot": [
            {
              "field": "numberSuccessful_ds",
              "value": 8,
              "count": 1
            },
            {
              "field": "numberSuccessful_ds",
              "value": 0,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 1,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 2,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 3,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 4,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 5,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 6,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 7,
              "count": 0
            },
            {
              "field": "numberSuccessful_ds",
              "value": 9,
              "count": 0
            }
          ]
        }
      ]
    }
  }
}
```

And I found the following result:
![Alt heatmap](http://s27.postimg.org/3ok3vu52r/Selection_044.png)

Which indicates it works well, so where is your problem?

**Comment by dougmcclure** [comment]
Ok, I think I've got a something here. Something appears to be off in how queries are passed and results returned when using a collection name versus a collection alias (what I'm using).

When I pass in a query URL which includes the very specific collection name, I see results in the browser's JSON display like I hared above.

When I pass in a query URL which includes my collection alias (which is default for all my banana dashboards), all results coming back as zero which explains why I do not see any count data in the heat map.

Any ideas on why this is behaving like it is?

**Comment by athana** [comment]
Hi @dougmcclure - I setup a test on my machine with Solr 4.7.1 in SolrCloud mode and use two shards. Then, I created two collections (hvac and hvac2) and one alias (hvacalias) that point to them. Then I create a Banana dashboard using 'hvacalias' and everything works fine. However, when I tried to add the Heatmap panel, it will come up empty at first because Solr 4.7.1 does not support distributed facet.pivot (https://issues.apache.org/jira/browse/SOLR-2894). And therefore, I have to go to Heatmap Settings => Queries tab => Add Customer Queries textbox, and add this:  &distrib=false. This will make the Heatmap panel (facet.pivot) works properly.

Is this the same setup in your environment? Are you running Banana inside your Solr server? Or you are accessing Solr on remote machines?

These are some screenshots from my test settings:

![heatmap_settings_queries](https://cloud.githubusercontent.com/assets/974546/6143296/ecc28332-b1f8-11e4-8843-8a53a0245a94.png)
![heatmap_settings_panel](https://cloud.githubusercontent.com/assets/974546/6143303/055517d4-b1f9-11e4-88b6-5b23f63035c5.png)
![test_dashboard](https://cloud.githubusercontent.com/assets/974546/6143320/491f5ba0-b1f9-11e4-88ed-fc097f237f21.png)
![dashboard_settings](https://cloud.githubusercontent.com/assets/974546/6143359/d7cbb02e-b1f9-11e4-9c23-3b6461ce79c1.png)

**Comment by dougmcclure** [comment]
Yes, exactly the same.  I have banana running inside the Solr server.
